tests/test_server.py

```
@@ -0,0 +1,417 @@
+        async with server:
+            pass
+        
+        server.start.assert_called_once()
+        server.stop.assert_called_once()
+
+    @pytest.mark.asyncio
+    async def test_server_relay_data(self, server, mocker):
+        """Test data relay between client and proxy."""
+        # This tests the _relay method if it exists
+        mock_reader1 = MagicMock()
+        mock_writer1 = MagicMock()
```

Ruff (0.11.9)

Local variable `mock_writer1` is assigned to but never used

Remove assignment to unused variable `mock_writer1`

(F841)